frontend/src/components/Profile/Profile.tsx

```diff
@@ -32,122 +45,162 @@ const InnerProfile: React.FC = () => {
     setOpen(false);
   };
 
+  useEffect(() => {
```

Comment:
We don't need this `useEffect` anymore after the new changes right?

Reply:
As we can edit, I think we need the useEffect

Reply:
This one will only run when `user` changes, which is handled whenever a page rerenders. @Imeth97 thoughts?

Reply:
I tried to run without the useEffect, it will not show the saved data once u refresh

Reply:
If you remove it, the formData will be set initially and won't be updated if the user data changes.

Reply:
Yep - got it. In that case can you refactor it like this then:

```useEffect(() => {
    if (user) {
      setFormData((previousFormData) => ({
        ...previousFormData,
        ...(!!user.user_metadata && user.user_metadata),
        ...(!!user.email && { email: user.email }),
      }));
    }
  }, [user]);```